### PR TITLE
Allow implementers to throw IOException

### PR DESCRIPTION
**Pull request by batmat**
@reviewbybees esp. @jglick 

downstream of https://github.com/jenkinsci/branch-api-plugin/pull/82

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -102,7 +102,8 @@ THE SOFTWARE.
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>branch-api</artifactId>
-            <version>2.0.2</version>
+            <!-- TODO: Remove when branch-api is released -->
+            <version>2.0.5-20170213.115501-2</version>
```
Comment (jglick) [inline pom.xml]: Is this downstream of some other PR? If so, mention it in this PR’s description, and in this comment.

File: src/main/java/org/jenkinsci/plugins/workflow/multibranch/AbstractWorkflowMultiBranchProjectFactory.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
         } // otherwise got recognized by something else before, oh well
     }
 
-    protected void customize(WorkflowMultiBranchProject project) {}
+    protected void customize(WorkflowMultiBranchProject project) throws IOException {}
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/multibranch/AbstractWorkflowMultiBranchProjectFactory.java]: May as well let it throw `InterruptedException` too.

File: pom.xml (hunk 1 above)
Comment (batmat) [inline pom.xml]: downstream of https://github.com/jenkinsci/branch-api-plugin/pull/82

File: src/main/java/org/jenkinsci/plugins/workflow/multibranch/AbstractWorkflowMultiBranchProjectFactory.java (hunk 2 above)
Comment (batmat) [inline src/main/java/org/jenkinsci/plugins/workflow/multibranch/AbstractWorkflowMultiBranchProjectFactory.java]: @jglick fixed.

**Comment by batmat** [comment]
Just updated the PR to now reference the just released 2.0.7 @jglick

**Pull request merged**